# StomaMicrobiota - Plasticity of the adult human small intestinal stoma microbiota

The script and data deposited in this repository are representative of the analysis that we have run in our manuscript which is published in Cell Host and Microbe in 2022. Link to manuscript: https://www.cell.com/cell-host-microbe/pdf/S1931-3128(22)00515-7.pdf  

The pipelines that were followed in this study are
- QIIME (http://qiime.org)
- Phyloseq (https://joey711.github.io/phyloseq/)
- MaAsLin2 (https://huttenhower.sph.harvard.edu/maaslin/)
- Kraken2 (https://ccb.jhu.edu/software/kraken2/)
- Braken (https://ccb.jhu.edu/software/bracken/)
- MIDAS (https://github.com/snayfach/MIDAS)
- InStrain (https://instrain.readthedocs.io/en/latest/)
- HAllA (https://huttenhower.sph.harvard.edu/halla/)
- ICRA and SGVFinder (https://github.com/segalab/SGVFinder/)

Data and Code Availability
All metagenomic sequencing data generated in this study has been deposited in https://www.ncbi.nlm.nih.gov/bioproject/ with BioProject Accession PRJNA825696 (Accession: SAMN27578394). Metabolomic raw dataset with processed files and the entire details of used samples can be downloaded using the following link: https://figshare.com/s/5f84cf321e731a49ae06 (DOI: 10.6084/m9.figshare.19483346). The 16S rRNA amplicon sequencing dataset with processed files and the entire details of used samples can be downloaded using the following link: https://figshare.com/s/077130437ca8ac314386 (DOI: 10.6084/m9.figshare.19483346). No new code was developed in this study. We just followed the developers' suggestions in our analysis with minor optimizations.
